admin/bin/check-ps-report-for-corrupt-data.js

```diff
@@ -0,0 +1,380 @@
+  if (val !== 'M' && val !== 'F') {
+    report(prop, row, val)
+  }
+}
+
+function checkPupilId (row, prop) {
+  const val = row[prop]
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (val.length !== 13) {
+    report(prop, row, val)
+  }
+}
+
+function checkForename (row, prop) {
+  const val = row[prop]
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (val.length < 1 || val.length > 128) {
+    report(prop, row, val)
+  }
+}
+
+function checkSurname (row, prop) {
+  const val = row[prop]
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (val.length < 1 || val.length > 128) {
+    report(prop, row, val)
+  }
+}
+
+function checkFormMark (row, prop) {
+  const val = Number(row[prop])
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (!(RA.isNumber(val)) && RA.inRange(0, 26, val)) {
+    report(prop, row, row[prop])
+  }
+}
+
+function checkQDisplayTime (row, prop) {
+  const val = Number(row[prop])
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (!RA.inRange(6, 9, val)) {
+    report(prop, row, row[prop])
+  }
+}
+
+function checkPauseLength (row, prop) {
+  const val = Number(row[prop])
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (val !== 3) {
+    report(prop, row, row[prop])
+  }
+}
+
+function checkDeviceType (row, prop) {
+  const val = row[prop]
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (val === '') {
+    report(prop, row, row[prop])
+  }
+}
+
+function checkDeviceTypeModel (row, prop) {
+  const val = row[prop]
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (val === '') {
+    report(prop, row, row[prop])
+  }
+}
+
+function checkDeviceID (row, prop) {
+  // DeviceId
+  const val = row[prop]
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (val === '') {
+    report(prop, row, row[prop])
+  }
+}
+
+function checkSchoolName (row, prop) {
+  // School Name
+  const val = row[prop]
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (val === '') {
+    report(prop, row, row[prop])
+  }
+}
+
+function checkEstab (row, prop) {
+  const regex = /^\d{4}$/
+  regexTest(row, prop, regex)
+}
+
+function checkSchoolURN (row, prop) {
+  const regex = /^\d{6}$/
+  regexTest(row, prop, regex)
+}
+
+function checkLAnum (row, prop) {
+  // should be LAnum
+  const regex = /^\d{3}$/
+  regexTest(row, prop, regex)
+}
+
+function checkAttemptID (row, prop) {
+  const regex = /^[0-9A-Za-z-]+$/
+  regexTest(row, prop, regex)
+}
+
+function checkFormID (row, prop) {
+  // should be FormID
+  const regex = /^[0-9A-Za-z]{4,}$/
+  regexTest(row, prop, regex)
+}
+
+function checkTestDate (row, prop) {
+  const regex = /^\d{8}$/
+  regexTest(row, prop, regex)
+}
+
+function checkTimeStart (row, prop) {
```

Comment:
these could also be merged to one that fits the whole group of properties to be checked.